### DISCUSSION TITLE
PowerShell import export xlsx :-(

### DISCUSSION THREAD

**Discussion by pmill06e** [dfinke/ImportExcel#1511] [Q&A] [answered]
Hello, brand new to powershell I made a small script which should take leaf1 from the file toto.xlsx and export by making an addition in the file tata.xlsx of leaf1. I don't have an error message but it doesn't copy anything in your opinion, what am I missing in this script so that the data is added correctly? thank you in advance. here my script : 
$cheminFichierSource = "C:\Users\pmill755\desktop\toto.xlsx"
$cheminFichierDestination = "C:\Users\pmill755\desktop\tata.xlsx"

#Charger les données de la feuille "Feuil1" du fichier source
$donneesSource = Import-Excel -Path $cheminFichierSource -WorksheetName "Feuil1"

#Charger les données de la feuille "Feuil1" du fichier destination (s'il existe)
if (Test-Path $cheminFichierDestination) {
    $donneesDestination = Import-Excel -Path $cheminFichierDestination -WorksheetName "Feuil1"
} else {
    # Créer un nouveau fichier tata.xlsx si le fichier destination n'existe pas encore
    $donneesDestination = @()
}

#Ajouter les données du fichier source aux données de destination
$donneesDestination += $donneesSource

#Exporter les données vers le fichier destination avec la feuille "Feuil1" en mode ajout
Export-Excel -Path $cheminFichierDestination -WorksheetName "Feuil1" -AutoSize -TableName "Donnees" -Append -NoHeader -PassThru -StartRow 1 -AutoFilter
#Afficher un message de confirmation
Write-Host "Les données ont été copiées avec succès vers le fichier [$cheminFichierDestination."[](url)](url)
[tata.xlsx](https://github.com/dfinke/ImportExcel/files/12606348/tata.xlsx)
[toto.xlsx](https://github.com/dfinke/ImportExcel/files/12606351/toto.xlsx)

**Comment by dfinke** [comment, marked as answer]
Thanks for using ImportExcel.

You are not piping your data to Export-Excel.

Try this. I would also recommend not using `NoHeader`. Needed for the `Autofilter` to work best.

```powershell
$cheminFichierDestination = "C:\Users\pmill755\desktop\report.xlsx"
Remove-Item $cheminFichierDestination -ErrorAction SilentlyContinue

$data = $(
    Import-Excel "C:\Users\pmill755\desktop\toto.xlsx"
    Import-Excel "C:\Users\pmill755\desktop\tata.xlsx"
)

$data | Export-Excel -Path $cheminFichierDestination -WorksheetName "Feuil1" -AutoSize -TableName "Donnees" -AutoFilter -Show
```

**Reply by pmill06e** [reply]
Thank you very much, it works now but I certainly expressed myself poorly on the need. Each time the script is launched, all of the records must be added to the output file, even if it is identical to the previous time. How to make the output file keep the date format for the Target Start and Target End fields.
Thank you again for your help

**Reply by pmill06e** [reply]
I just made it work with adding data each time the script is used. First I added this command $combinedData = $existingData + $data. I had an error: $DestinationFilePath -WorksheetName "Sheet1" -TableName "Data …
     | ~~~~~~~~~~
     | A parameter cannot be found that matches parameter name 'TableName' I removed TableName and now it works. Do you have an explanation on the TableName problem or my way of doing things? Thank you for enlightening my brain

**Reply by dfinke** [reply]
That's more involved and I'd need time as well as a bit more understanding.

The `-Append` is needed most likely. You did have PassThru, which you need to set to a variable then there are functions like `Set-ExcelRange` that take a worksheet, a range if cells and then you can format numbers etc, etc.

It's more involved, if that is in fact what you need.

I'm not sure why date format would not stay the same.

**Reply by pmill06e** [reply]
Hello, don't waste time, what I manage to do suits me and it's thanks to you. I wish you a good day
